### PR TITLE
pythonPackages.thefuzz: init at 0.19.0

### PR DESCRIPTION
**Pull request by sumnerevans**
###### Description of changes

https://github.com/seatgeek/fuzzywuzzy was renamed to https://github.com/seatgeek/thefuzz

I'm not really sure how to handle the deprecation, since the package name change does require users of the package to change their code, so I think it makes sense to just include it as a separate package.

Signed-off-by: Sumner Evans <me@sumnerevans.com>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sumnerevans** [comment]
pinging @erikarvstedt, maintainer of the fuzzywuzzy package

File: pkgs/development/python-modules/thefuzz/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ lib, buildPythonPackage, fetchPypi, python-Levenshtein, pycodestyle, hypothesis, pytest }:
```
Comment (fabaff) [inline pkgs/development/python-modules/thefuzz/default.nix:1]: One item per line make diffs in the future easier to read.

```suggestion
{ lib
, buildPythonPackage
, fetchPypi
, python-levenshtein
, pycodestyle
, hypothesis
, pytest
}:
```

File: pkgs/development/python-modules/thefuzz/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+buildPythonPackage rec {
+  pname = "thefuzz";
+  version = "0.19.0";
```
Comment (fabaff) [inline pkgs/development/python-modules/thefuzz/default.nix:5]: New packages should be limited to supported Python releases as the chance is high that they are not used/tested on deprecated releases.

```suggestion
  version = "0.19.0";
  format = "setuptools";

  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/thefuzz/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-b3Em2y8silQhKwXjp0DkX0KRxJfXXSB1Fyj2Nbt0qj0=";
+  };
+
+  propagatedBuildInputs = [ python-Levenshtein ];
```
Comment (fabaff) [inline pkgs/development/python-modules/thefuzz/default.nix:12]: `python-levenshtein` is an [extra dependency](https://github.com/seatgeek/thefuzz/blob/master/setup.py#L26).

```
  passthru.optional-dependencies = {
    speedup = [
      python-levenshtein
    ];
  };
```

File: pkgs/development/python-modules/thefuzz/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  checkInputs = [ pycodestyle hypothesis pytest ];
```
Comment (fabaff) [inline pkgs/development/python-modules/thefuzz/default.nix:13]: Patch out the `pycodestyle` requirement as we don't want to lint code. Please check if it's possible to use 

```suggestion
  checkInputs = [
    hypothesis
    pytestCheckHook
  ];
```

File: pkgs/development/python-modules/thefuzz/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+  meta = with lib; {
+    description = "Fuzzy string matching for Python";
+    homepage = "https://github.com/seatgeek/thefuzz";
```
Comment (fabaff) [inline pkgs/development/python-modules/thefuzz/default.nix:17]:
```suggestion
    homepage = "https://github.com/seatgeek/thefuzz";
    changelog = "https://github.com/seatgeek/thefuzz/blob/${version}/CHANGES.rst";
```

File: pkgs/development/python-modules/thefuzz/default.nix [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    license = licenses.gpl2;
```
Comment (fabaff) [inline pkgs/development/python-modules/thefuzz/default.nix:18]: No statement that later version of the license are supported.

```suggestion
    license = licenses.gpl2Only;
```

**Review by fabaff — COMMENTED** [review]
At first glance it doesn't look like that the renaming will be handled by PyPI as the  `fuzzywuzzy` module is not pointing to `thefuzz`.

We could handle it with an alias for us but that would not really help. With 0.19.0 the imports changed.

File: pkgs/development/python-modules/thefuzz/default.nix (hunk 3 above)
Comment (sumnerevans) [reply pkgs/development/python-modules/thefuzz/default.nix:12]: how would one enable this optional dependency when installing the package?

Comment (erikarvstedt) [reply pkgs/development/python-modules/thefuzz/default.nix]: @sumnerevans, see section 17.27.2.3.2 in the [nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#reference).

@fabaff python-levenshtein (a C extension) seems to work on all platforms that support Python. Is there any downside to enabling it by default?

File: pkgs/development/python-modules/thefuzz/default.nix [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+, hypothesis
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "thefuzz";
+  version = "0.19.0";
+  format = "setuptools";
+
+  disabled = python3.pkgs.pythonOlder "3.7";
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/thefuzz/default.nix:16]: why not to be consistent and

```suggestion
  disabled = pythonOlder "3.7";
```

and add it in the inputs list?

File: pkgs/top-level/python-packages.nix [hunk 8]
```diff
@@ -11199,6 +11199,8 @@ self: super: with self; {
     cudnnSupport = false;
   };
 
+  thefuzz = callPackage ../development/python-modules/thefuzz { };
```
Comment (AndersonTorres) [inline pkgs/top-level/python-packages.nix:11202]: Being such a Python package, I would suggest the Python's `callPackage`.
Hovever I am a bit lost on this.

File: pkgs/development/python-modules/thefuzz/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/python-modules/thefuzz/default.nix]: I think that `thefuzz` should work the same way as `fuzzywuzzy and if most user would enable it anyway then it makes perfect sense to enable it by default.

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
[Fixup](https://github.com/erikarvstedt/nixpkgs/commits/thefuzz).

Fetch with:
```bash
git fetch https://github.com/erikarvstedt/nixpkgs thefuzz
git show FETCH_HEAD
```

**Review by AndersonTorres — APPROVED** [review]

**Review by erikarvstedt — APPROVED** [review]
- [x] Built on `x86_64-linux`

**Pull request merged**